### PR TITLE
feat(eks): Auto select AMI type for T4g instance type 

### LINKED ISSUE #10361: [aws-eks] support T4g instances
New EC2 T4g Instances – Burstable Performance Powered by AWS Graviton2 – Try Them for Free
https://aws.amazon.com/tw/blogs/aws/new-t4g-instances-burstable-performance-powered-by-aws-graviton2/

### Proposed Solution

Add `t4g` in the graviton2 instance types

https://github.com/aws/aws-cdk/blob/25256d09aa39b11f49bcb0479fbe2ec2bc1e145d/packages/%40aws-cdk/aws-eks/lib/instance-types.ts#L5

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by pahud**
Closes #10361 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by pahud** [comment]
Hi @iliapolo 

This is weird. The build was SUCCESS but the CI failed. Is it possible to trigger the rebuild without pushing any new commit?

**Comment by pahud** [comment]
@iliapolo  I am ready :-)

File: packages/@aws-cdk/aws-eks/test/test.cluster.ts [hunk 1]
```diff
@@ -1162,6 +1162,26 @@ export = {
       test.done();
     },
 
+    'addNodegroup with T4g instance type comes with nodegroup with correct AmiType'(test: Test) {
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-eks/test/test.cluster.ts:1165]: Also add a test for the auto selection of the cpu arch when creating an image for auto-scaling group capacity:

https://github.com/aws/aws-cdk/blob/028495e55fed02f727024a443fc29a17d4629fe3/packages/%40aws-cdk/aws-eks/lib/cluster.ts#L1795-L1799

**Review by iliapolo — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-eks/test/test.cluster.ts (hunk 1 above)
Comment (pahud) [reply packages/@aws-cdk/aws-eks/test/test.cluster.ts]: DONE

**Comment by pahud** [comment]
@iliapolo please check it out again.

**Review by iliapolo — APPROVED** [review]

**Pull request merged**